packages/wonder-blocks-popover/components/popover-anchor.test.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import * as React from "react";
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
+import PopoverAnchor from "./popover-anchor.js";
+
+describe("PopoverAnchor", () => {
+    afterEach(() => {
+        unmountAll();
+    });
+
+    it("should set child node as ref", () => {
+        // Arrange
+        const updateRef = jest.fn();
+
+        const wrapper = mount(
+            <PopoverAnchor anchorRef={updateRef} onClick={jest.fn()}>
+                <button data-anchor>test</button>
+            </PopoverAnchor>,
+        );
+
+        // Act
+        const triggerElement = wrapper.find("[data-anchor]").getDOMNode();
+
+        // Assert
+        expect(updateRef).toBeCalledWith(triggerElement);
+    });
+
+    it("should add onClick handler if child is a function", () => {
+        // Arrange
+        const onClickMock = jest.fn();
+
+        const wrapper = mount(
+            <PopoverAnchor anchorRef={jest.fn()} onClick={onClickMock}>
+                {({open}) => (
+                    <button onClick={open} data-anchor>
+                        open
+                    </button>
+                )}
+            </PopoverAnchor>,
+        );
+
+        // Act
+        wrapper.find("[data-anchor]").simulate("click");
+
+        // Assert
+        expect(onClickMock).toBeCalled();
+    });
+
+    it("should add onClick handler if child is a Node", () => {
+        // Arrange
+        const onClickMock = jest.fn();
+        const onClickInnerMock = jest.fn();
+
+        const wrapper = mount(
+            <PopoverAnchor anchorRef={jest.fn()} onClick={onClickMock}>
+                <button onClick={onClickInnerMock} data-anchor>
+                    test
+                </button>
+            </PopoverAnchor>,
+        );
+
+        // Act
+        wrapper.find("[data-anchor]").simulate("click");
+
+        // Assert
+        // both custom and internal click should be called
+        expect(onClickInnerMock).toBeCalled();
+        expect(onClickMock).toBeCalled();
```

Comment:
I know it's repetitive, but I'd rather each expect had it's own test case. They are verifying different things and if the first expectation fails, we never test the second, which means that we lose fidelity in our test runs. If they are separate test cases then we get more information from our test runs.

Could you split these into separate test cases please?

Reply:
Agreed that it's ideally better to use one `expect` per test, but I think this is a special case where we want to test that the inner `onClick` is preserved and called at the same time with the PopoverAchor's `onClick` handler.

I separated this into two different tests, but maybe the second test feels kind of redundant (?), as we would be testing implementation details in this case? We would be testing that the onClick handler gets called if the button is clicked.

This would be result of creating the second test:
```
it("should also call the internal onClick handler if child is a Node", () => {
    // Arrange
    const onClickInnerMock = jest.fn();

    const wrapper = mount(
        <PopoverAnchor anchorRef={jest.fn()} onClick={jest.fn()}>
            <button onClick={onClickInnerMock} data-anchor>
                test
            </button>
        </PopoverAnchor>,
    );

    // Act
    wrapper.find("[data-anchor]").simulate("click");

    // Assert
    expect(onClickInnerMock).toBeCalled();
});
```

What do you think? Do you find a better way to approach this test?